client/client.go

```diff
@@ -311,6 +312,9 @@ func (pc *PolkadotClient) GetMetadata(ctx context.Context, sess structs.APISessi
 	if err != nil {
 		return nil, err
 	}
+	if resp[0].Result == nil {
```

Comment:
```suggestion
	if len(resp) == 0 || resp[0].Result == nil {
```
It looks like `Send` can return a nil response for which `len(resp)` would evaluate to zero. I'm not sure if this warrants changing the error message or not.

Reply:
Actually don't think this can happen if the error returned is nil, so it is up to you if you think the check should be added.

Reply:
Yes, I am not 100% sure yet if this is a good or needed change..  I mentioned in my comments that I am starting to think we are using a non thread safe library which is causing some strange errors in local testing.